src/cli/Extract.cpp

```diff
@@ -55,15 +61,25 @@ int Extract::execute(QStringList arguments)
         return EXIT_FAILURE;
     }
 
-    out << "Insert the database password\n> ";
-    out.flush();
-
     CompositeKey compositeKey;
 
+    QString keyFilePath = parser.value(keyFile);
+    FileKey fileKey;
+    QString errorMsg;
+    if (!fileKey.load(keyFilePath, &errorMsg)) {
+        qCritical("Failed to load file key %s: %s",
+                  qPrintable(keyFilePath), qPrintable(errorMsg));
+        return EXIT_FAILURE;
+    }
```

Comment:
@yan12125 shouldn't all that code be executed only if the `key-file` option was used?

```
if (!parser.value(keyFile).isEmpty()) {
...
}
```
Otherwise you're enforcing the use of a key file.

the `compositeKey.addKey(fileKey);` line could also be moved in the if